var/spack/repos/builtin/packages/py-smartredis/package.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class PySmartredis(PythonPackage):
+    """A Python Interface for the SmartRedis Library Client"""
+
+    homepage = "https://www.craylabs.org/docs/smartredis.html"
+    pypi = "smartredis/smartredis-0.4.0.tar.gz"
+    git = "https://github.com/CrayLabs/SmartRedis"
+
+    maintainers("MattToast")
+
+    version("0.4.1", sha256="fff16ed1eb09648ac3c3f845373beb37f3ffe7414d8745ae36af9daf585f8c5b")
+    version("0.4.0", sha256="d12779aa8bb038e837c25eac41b178aab9e16b729d50ee360b5af8f813d9f1dd")
+
+    depends_on("python@3.8:3.10", type=("build", "run"))
+    depends_on("py-setuptools@42:", type=("build",))
+    depends_on("cmake@3.13:", type=("build",))
+    depends_on("hiredis@1.0:", type=("build", "link", "run"))
+    depends_on("redis-plus-plus@1.3: cxxstd=17", type=("build", "link"))
```

Comment:
Where did you get this versions from?

According to [v0.4.0#dependencies](https://github.com/CrayLabs/SmartRedis/tree/v0.4.0#dependencies)  shouldn't they be:
- Hiredis 1.0.0
- Redis-plus-plus 1.2.3

and for [v0.4.1#dependencies](https://github.com/CrayLabs/SmartRedis/tree/v0.4.1#dependencies):
- Hiredis 1.1.0
- Redis-plus-plus 1.3.5

Reply:
Whoops! I know from experience that the required dependencies of SR are a bit looser than they are listed in the docs. While experimenting, I just threw some random versions up that I knew were "close enough" to get something working w/o consulting the docs.

Definitely should be changed! Will pin to the appropriate versions!!

Reply:
Follow up question: I noticed that `redis-plus-plus` v1.2.3 is not currently available through spack. Given that `redis-plus-plus` follows semantic versioning, v1.3.5 should be a compatible dependency, so I have tied both SR v0.4.1 and 0.4.0 to that version for now.

Do let me know if you think this is an appropriate solution, or if we should go another route of either:
1) separately adding a new v1.2.3 version of a `redis-plus-plus`
2) removing the SR v0.4.0 version from this package